File: lua/pe009.lua
-- this computes projecteuler.net problem 009

stop = 1000
for a = 1,stop do
  for b = a+1,stop do
    for c = b+1,stop do
      if a+b+c == stop then
        if (a*a)+(b*b) == (c*c) then
          print(a*b*c)
	  os.exit(0)
	end
      end
    end
  end
end
os.exit(1)

File: python/pe009.py
#!/usr/bin/env python3

""" this computes projecteuler.net problem 009 """

import sys

STOP = 1000
for a in range(1, STOP):
    for b in range(a + 1, STOP):
        for c in range(b + 1, STOP):
            if a + b + c == STOP:
                if (a * a) + (b * b) == (c * c):
                    print(a * b * c)
                    sys.exit(0)
sys.exit(1)
